code/modules/reagents/reagent_dispenser.dm

```diff
@@ -1,196 +1,340 @@
 /obj/structure/reagent_dispensers
-	name = "Dispenser"
-	desc = "..."
-	icon = 'icons/obj/objects.dmi'
-	icon_state = "water"
+	name = "storage tank"
+	desc = "It can hold large amount of chemicals."
+	icon = 'icons/obj/chemtank.dmi'
+	icon_state = "chemtank"
 	density = 1
 	anchored = 0
 	pressure_resistance = 2*ONE_ATMOSPHERE
-	obj_integrity = 300
-	max_integrity = 300
-	var/tank_volume = 1000 //In units, how much the dispenser can hold
-	var/reagent_id = "water" //The ID of the reagent that the dispenser uses
+	var/list/brokenvars = list("storage tank", "chemtank", /obj/structure/reagent_dispensers)// when it blows up,what will it become,made it qdel src and make a statuelike thing to avoid having to change syringes,droppers,glasses etc with snowflakey checks
+	var/exploded = 0 // if atleast one reaction happened in boom(), this becomes 1
+	var/list/bypassUIicons = list( // list which have all the icons where reagent overlay and update_icon proc in general must not work
+		"peppertank",
+		"water_cooler",
+		"beerkeg",
+		"virusfoodtank",
+		"honk_cooler",
+	)
 
-/obj/structure/reagent_dispensers/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir)
-	. = ..()
-	if(. && obj_integrity > 0)
-		if(tank_volume && (damage_flag == "bullet" || damage_flag == "laser"))
-			boom()
+/obj/structure/reagent_dispensers/on_reagent_change()
+	update_icon()
 
-/obj/structure/reagent_dispensers/attackby(obj/item/weapon/W, mob/user, params)
-	if(istype(W, /obj/item/weapon/screwdriver))
-		if(src.icon_state == "water" && !src.is_open_container())
-			src.icon_state = "wateropen"
-			user << "<span class='notice'>You unscrew and remove the lid from [src].</span>"
-			src.container_type |= OPENCONTAINER
-			src.update_icon()
-			playsound(src, 'sound/items/Deconstruct.ogg', 20, 1)
-			desc = "A water tank. It's top has been removed."
-			return
-		if(src.icon_state == "wateropen" && src.is_open_container())
-			src.icon_state = "water"
-			user << "<span class='notice'>You reattach and screw the [src]'s lid into place.</span>"
-			src.container_type &= ~OPENCONTAINER
-			src.update_icon()
-			playsound(src, 'sound/items/Deconstruct.ogg', 20, 1)
-			desc = "A water tank. It has a top that can be removed."
-			return
-		else
-			user << "<span class='warning'>This device does not have a removable lid!</span>"
-			return
-	if(istype(W, /obj/item/weapon/reagent_containers))
-		return 0 //so we can refill them via their afterattack.
-	else
-		return ..()
-
-/obj/structure/reagent_dispensers/New()
-	create_reagents(tank_volume)
-	reagents.add_reagent(reagent_id, tank_volume)
+/obj/structure/reagent_dispensers/attack_hand()
 	..()
+	update_icon()
 
-/obj/structure/reagent_dispensers/examine(mob/user)
-	..()
+/obj/structure/reagent_dispensers/update_icon()
+	overlays.Cut()

```

Comment:
**_Please do not manually change the overlays_**

Use `/proc/cut_overlays()` and `/proc/add_overlay(image, priority = 0)` so that overlay changes do not cause any lag